tawazi/dag.py

```diff
@@ -414,23 +487,224 @@ def get_highest_priority_nodes(nodes: List[ExecNode]) -> List[ExecNode]:
                     wait(futures.values(), return_when=ALL_COMPLETED)
         return node_dict
 
+    def _get_leaves_ids(
+        self, twz_nodes: List[Union[Tag, IdentityHash, ExecNode]]
+    ) -> List[IdentityHash]:
+        """
+        get the ids of ExecNodes corresponding to twz_nodes.
+        The identification can be carried out using the tag, the Id, or the ExecNode itself.
+        Keep in Mind that depending on the way ExecNode is provided inside twz_nodes,
+         the returned id
+
+        Args:
+            twz_nodes (List[Union[Tag, IdentityHash, ExecNode]]): list of a mix of identifier that the user might provide to run a subgraph
+
+        Raises:
+            ValueError: if a requested ExecNode is not found in the DAG
+            TawaziTypeError: if the Type of the identifier is not Tag, IdentityHash or ExecNode
+            TawaziBaseException: if the returned List[IdentityHash] has the wrong length, this indicates a bug in the code
+
+        Returns:
+            List[IdentityHash]: ExecNodes' Identities
+        """
+        leaves_ids = []
+        for tag_or_id_or_node in twz_nodes:
+            if isinstance(tag_or_id_or_node, ExecNode):
+                leaves_ids.append(tag_or_id_or_node.id)
+            # todo: do further validation!
```

Comment:
in case tag_or_id_or_node is a tag, we should verify that it is either a string or tuple or immutable.. The allowed types are not clear as of yet so I decided to do this validation later

Reply:
yeah it makes sense, I'll close